### PR TITLE
MAYA-121242 fix flipped normals.

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
Maya did not write the USD "orientation" attribute. In addition, Maya flipped the vertices order when editing a USD mesh with "leftHanded" orientation. On merge-to-USD, Maya would write right-handed vertices but not write the "orientation" property.

The result would be righ-handed verticeswith left-handed orientation attribute.

See: MAYA-105623 for the original reason we flip vertices orders. It has to do with the way "leftHanded" affect vertices order and normals in USD.

The fix is to explicitly write the "orientation" attribute on export.

A few improvements in the verbosity options and simplification of some code were also made whilte investigating the problem and are included.

### CODE REVIEW

File: lib/usd/utils/MergePrims.cpp [hunk 1]
```diff
@@ -249,19 +249,35 @@ bool isLocalTransformModified(const UsdPrim& srcPrim, const UsdPrim& dstPrim)
 // Special metadata handling.
 //
 // There are some metadata that we know the merge-to-USD temporary export-to-USD
-// will never produce. List them here to be always preserved.
+// will never produce. Others are generated only when needed, so their absence
+// should always means to remove them.
```
Comment (ppt-adsk) [inline lib/usd/utils/MergePrims.cpp:253]: "should always means" --> "should always mean".

**Review by ppt-adsk — APPROVED** [review]

**Comment by dj-mcg** [comment]
Hi @pierrebai-adsk & @kxl-adsk - This broke one of our round trip tests I think. We start with a .ma file that contains a mesh with leftHanded orientation, which is exported to USD properly. Then we re-import that same exported USD file, then re-export. This re-exported USD file flips the verts and authors rightHanded orientation. 

Is this behavior expected? I assumed it should properly import (and re-export) the leftHanded USD mesh but I don't have the context that you mentioned regarding Maya flipping the vert order.

Thanks,
Dan

**Comment by dj-mcg** [comment]
Also, is there a reason we don't write the orientation attribute sparsely? It looks like it's alwasy explicitly authoring opinions with the default value (rightHanded). I'm happy to make that change unless there's a reason not to?

**Comment by pierrebai-adsk** [comment]
The problem was that the "opposite" flag in Maya flips the vertex order of the face *and* the normals, while the one in USD only flips the face order. So in Maya we cannot use "opposite" to only do one thing. So, when we encounter leftHanded in USD, we cannot use "opposite".

Maybe in the future we could undo the vertex re-ordering on export. We would need to keep around something that tells us to do so.

**Comment by pierrebai-adsk** [comment]
I'll add that we recently improved the transformation round-trip by using a notion of "equivalent transform" that let us detect that two transforms might be build up differently but represent the same value. We could do something similar for vertices and normals, taking into consideration handed-ness when comparing the raw value arrays. This would at least detect that they have not been edited and avoid authoring opinions.

This is all speculative work though, we'd need to schedule and implement it.

**Pull request merged**